scripts/zones/Port_Jeuno/npcs/Illauvolahaut.lua

```diff
@@ -15,12 +15,16 @@ end
 entity.onTrigger = function(player, npc)
     local kazhamPass = player:hasKeyItem(xi.ki.AIRSHIP_PASS_FOR_KAZHAM)
 
-    if not kazhamPass then
-        player:startEvent(35) -- without pass
-    elseif kazhamPass and player:getGil() < 200 then
-        player:startEvent(45) -- Pass without money
-    elseif kazhamPass then
-        player:startEvent(37) -- Pass with money
+    if player:getCharVar("Quest[0][67]AIRSHIP_BLOCK") > os.time() then
+            player:startEvent(41)
```

Comment:
add to quest framework?

Reply:
can't because the quest is deleted when the player fails, and this dialogue remains until japanese midnight. Nor does the quest become available until JP midnight

Reply:
could be added to framework in the a block 
```lua
{
        check = function(player, status, vars)
            return status == QUEST_AVAILABLE and
                quest:getVar(player, 'Stage') > os.time()
        end,
```

for all the npcs using this logic